src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -283,6 +289,65 @@
             return this.itemsContainer;
         }
 
+        hasChangedOptions(mutationList) {
+            return mutationList.reduce(
+                (output, mutationRecord) => output || mutationRecord.addedNodes.length || mutationRecord.removedNodes.length,
+                false,
+            );
+        }
+
+        getSelectedItems() {
+            if (this.canSelectOnlyOne) {
+                const selectedItem = this.sourceInput.querySelector(`[value="${this.sourceInput.value}"]`);
+
+                return selectedItem ? [selectedItem] : [];
+            }
+
+            return [...this.sourceInput.querySelectorAll('.ibexa-input:checked')];
+        }
+
+        recreateOptions() {
+            const newOptions = this.sourceInput.querySelectorAll('option');
+
+            this.itemsListContainer.querySelectorAll('.ibexa-dropdown__item').forEach((item) => {
+                this.removeOption(item.dataset.value);
+            });
+
+            newOptions.forEach((option) => {
+                this.createOption(option.value, option.innerHTML);
+            });
+
+            const selectedItems = this.getSelectedItems();
+
+            this.clearCurrentSelection();
```

Comment:
`clearCurrentSelection` is not enough - `fitItems` should be called to adapt `ibexa-dropdown__selected-overflow-number` etc.